virttest/utils_test/libvirt.py

```diff
@@ -2846,9 +2846,24 @@ def set_vm_disk(vm, params, tmp_dir=None, test=None):
         If disk_src_name is given, replace current source file
         Otherwise, use current source file with update params.
         """
-        if disk_src_name:
-            blk_source = disk_src_name
-        disk_params_src = {"source_file": blk_source}
+        if disk_type == "block":
```

Comment:
Your code does not match the comment above -- 'If disk_src_name is given, replace current source file
￼        Otherwise, use current source file with update params.'

What about adding a new 'elif' for this type?